tests/e2e/streams/streams_tests.py

```diff
@@ -342,85 +367,137 @@ def test_check_already_started_stream(topics, connection):
+        f"TOPICS {','.join(topics)} "
+        f"TRANSFORM {transformation} "
+        f"BOOTSTRAP_SERVERS '{local}'",
+    )
     common.start_stream(cursor, "test")
     time.sleep(5)
 
     for topic in topics:
         producer.send(topic, SIMPLE_MSG).get(timeout=60)
 
     for topic in topics:
-        common.check_vertex_exists_with_topic_and_payload(
-            cursor, topic, SIMPLE_MSG)
+        common.check_vertex_exists_with_topic_and_payload(cursor, topic, SIMPLE_MSG)
 
 
 @pytest.mark.parametrize("transformation", TRANSFORMATIONS_TO_CHECK)
 def test_bootstrap_server_empty(producer, topics, connection, transformation):
     assert len(topics) > 0
     cursor = connection.cursor()
     with pytest.raises(mgclient.DatabaseError):
-        common.execute_and_fetch_all(cursor,
-                                     "CREATE STREAM test "
-                                     f"TOPICS {','.join(topics)} "
-                                     f"TRANSFORM {transformation} "
-                                     "BOOTSTRAP_SERVERS ''")
+        common.execute_and_fetch_all(
+            cursor,
+            "CREATE STREAM test "
+            f"TOPICS {','.join(topics)} "
+            f"TRANSFORM {transformation} "
+            "BOOTSTRAP_SERVERS ''",
+        )
+
+
+@pytest.mark.parametrize("transformation", TRANSFORMATIONS_TO_CHECK)
+def test_set_offset(producer, topics, connection, transformation):
+    assert len(topics) > 0
+    cursor = connection.cursor()
+    common.execute_and_fetch_all(
+        cursor,
+        "CREATE STREAM test "
+        f"TOPICS {topics[0]} "
+        f"TRANSFORM {transformation} "
+        "BATCH_SIZE 1",
+    )
+
+    messages = [f"{i} message" for i in range(1, 21)]
+
+    for message in messages:
+        producer.send(topics[0], message.encode()).get(timeout=60)
+
+    def execute_set_offset_and_consume(id):
+        common.execute_and_fetch_all(
+            cursor, f"CALL mg.set_stream_offset('test', {id}) YIELD result"
+        )
+        common.start_stream(cursor, "test")
+
+        time.sleep(2)
+        common.stop_stream(cursor, "test")
+        res = common.execute_and_fetch_all(cursor, "MATCH (n) RETURN n.payload")
+        return res
+
+    with pytest.raises(mgclient.DatabaseError):
+        res = common.execute_and_fetch_all(
+            cursor, f"CALL mg.set_stream_offset('foo', 10) YIELD result"
+        )
+
+    res = execute_set_offset_and_consume(10)
+    assert len(res) == 10
+    comparison_check = lambda a, b: a == str(b).strip("'(,)")
+    assert all([comparison_check(a, b) for a, b in zip(messages[10:], res)])
+    common.execute_and_fetch_all(cursor, "MATCH (n) DETACH DELETE n")
+
+    res = execute_set_offset_and_consume(-1)
+    assert len(res) == 20
+
+    assert all([comparison_check(a, b) for a, b in zip(messages, res)])
+    res = common.execute_and_fetch_all(cursor, "MATCH (n) DETACH DELETE n")
+
+    res = execute_set_offset_and_consume(-2)
+    assert len(res) == 0
```

Comment:
Consider sending one more message to verify the stream can consume message after setting the offset to the latest.

Reply:
Good idea, done!